### PR TITLE
Distinct subdirectory in docs/README.md

### PR DESCRIPTION
**Pull request by qin-nz**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ The code and documentation are hosted in the following repository: https://githu
 ```
 p4c
 ├── build                     -- recommended place to build binary
-├── p4include                 -- standard P4 files needed by the compiler
+├── p4include                 -- standard P4 files needed by the compiler (e.g., core.p4)
 ├── backends
```
Comment (ChrisDodd) [inline docs/README.md]: Keep this in alphabetical order?  Makes it easier to avoid accidental duplication.

**Comment by qin-nz** [comment]
@hanw I am a student in BUPT, who is a institutional member of P4.org.

File: docs/README.md [hunk 2]
```diff
@@ -8,23 +8,24 @@ The code and documentation are hosted in the following repository: https://githu
...
 │   └── doxygen               -- documentation generation support
+├── extensions
+│   └── XXXX                  -- symlinks to custom back-ends
 ├── frontends
 │   ├── common                -- common front-end code
 │   ├── p4-14                 -- P4_14 front-end
 │   └── p4                    -- P4_16 front-end
 ├── ir                        -- core internal representation
 ├── lib                       -- common utilities (libp4toolkit.a)
+├── m4
```
Comment (jnfoster) [inline docs/README.md]: M4 is a preprocessing system. See http://www.gnu.org/software/m4/m4.html.

**Comment by mbudiu-vmw** [comment]
@hanw: Maybe we can accept this one?

**Pull request merged**